packages/flutter/lib/src/widgets/dismissible.dart

```diff
@@ -499,6 +528,11 @@ class _DismissibleState extends State<Dismissible> with TickerProviderStateMixin
         _startResizeAnimation();
       else
         _moveController!.reverse();
+        if(widget.onDismissUpdate != null) {
+          final bool oldDismissThresholdReached = _dismissThresholdReached;
+          _dismissThresholdReached = false;
+          widget.onDismissUpdate!(_dismissDirection, DismissUpdate(reached: _dismissThresholdReached, previousReached: oldDismissThresholdReached));
+        }
```

Comment:
Either add `{ }` to the `if...else` bodies or correctly indent the new code. (Also should this have been caught by the unit tests?

Reply:
This was indeed wrong, did not notice that. There was a bigger issue here so I changed where the _dismissThresholdReached is calculated by adding a listener to the AnimationController, if that's ok. I also added a test now that would catch this.
Previously, when confirmDismiss returned false, the values would not have been updated at all.